src/transforms/pipelines/filter.rs

```diff
@@ -17,56 +15,54 @@ use serde::{Deserialize, Serialize};
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct PipelineFilterConfig {
     condition: AnyCondition,
-    inner: Box<dyn TransformConfig>,
 }
 
 impl PipelineFilterConfig {
-    pub fn new(condition: AnyCondition, inner: Box<dyn TransformConfig>) -> Self {
-        Self { condition, inner }
-    }
-
-    fn truthy_path(&self) -> Box<dyn TransformConfig> {
-        let mut result: IndexMap<String, Box<dyn TransformConfig>> = IndexMap::new();
-        result.insert(
-            "filter".to_string(),
-            Box::new(FilterConfig::from(self.condition.clone())),
-        );
-        result.insert("transforms".to_string(), self.inner.clone());
-        Box::new(ExpanderConfig::serial(result))
-    }
-
-    fn falsy_path(&self) -> Box<dyn TransformConfig> {
-        Box::new(FilterConfig::from(AnyCondition::Map(Box::new(
-            NotConfig::from(self.condition.clone()),
-        ))))
+    pub const fn new(condition: AnyCondition) -> Self {
+        Self { condition }
     }
 }
 
 #[async_trait::async_trait]
 #[typetag::serde(name = "pipeline_filter")]
 impl TransformConfig for PipelineFilterConfig {
-    async fn build(&self, _context: &TransformContext) -> crate::Result<Transform> {
-        Err("this transform must be expanded".into())
+    async fn build(&self, context: &TransformContext) -> crate::Result<Transform> {
+        Ok(Transform::DispatchFunction(Box::new(PipelineFilter {
+            condition: self.condition.build(&context.enrichment_tables)?,
+        })))
     }
 
-    fn expand(
-        &mut self,
-    ) -> crate::Result<Option<(IndexMap<String, Box<dyn TransformConfig>>, ExpandType)>> {
-        let mut result = IndexMap::new();
-        result.insert("truthy".to_string(), self.truthy_path());
-        result.insert("falsy".to_string(), self.falsy_path());
-        Ok(Some((result, ExpandType::Parallel { aggregates: true })))
+    fn named_outputs(&self) -> Vec<(Option<String>, DataType)> {
+        vec![
+            (Some("truthy".to_owned()), DataType::Any),
+            (Some("falsy".to_owned()), DataType::Any),
```

Comment:
The types here seem like a regression from the previous behavior. We used to get proper type checking because we knew the type of the inner transform, but now we don't. It's not the end of the world, but it'd be nice to figure out a way to keep that information.